### DISCUSSION TITLE
Behavior of 'n_startup_trials' in TPESampler

### DISCUSSION THREAD

**Discussion by vkyprmr** [optuna/optuna#4069] [Q&A]
Hello everyone,

Firstly, I would like to mention that I searched for this issue, found similar discussions but didn't find an answer. Hence, I am asking it again. I want to understand how things work under the hood.

This is my first time trying the `TPESampler`. In my used case, I want  to make the results from Optuna as reproducible as possible. Hence, I tried setting `seed` and `n_startup_trials=0`. However, this setting throws me the following warning and then the search gets stuck sometimes and sometimes it continues (mostly when n_startup_trials > 1):
```
RuntimeWarning: invalid value encountered in divide
  vecs = (vecs - lb) / (ub - lb)
```
No `error` is thrown. The same situation also happens with `n_startup_trials=1`.

Can someone help me understand why this is the case, or can I even set `n_startup_trials` to 0 or 1? If not, should somebody add it to the documentation?

**Comment by not522** [comment]
I think this bug was already fixed by #3953. Please try the latest Optuna.
